infrastructure/src/main/kotlin/KotlinLibraryPlugin.kt

```diff
@@ -1,34 +1,39 @@
 package com.redmadrobot.build
 
-import com.redmadrobot.build.extension.redmadrobotExtension
+import com.redmadrobot.build.extension.TestOptions
+import com.redmadrobot.build.internal.*
 import org.gradle.api.JavaVersion
-import org.gradle.api.Plugin
 import org.gradle.api.Project
+import org.gradle.api.tasks.testing.Test
 import org.gradle.kotlin.dsl.apply
 import org.gradle.kotlin.dsl.repositories
+import org.gradle.kotlin.dsl.withType
 
-public class KotlinLibraryPlugin : Plugin<Project> {
-    override fun apply(target: Project) {
-        with(target) {
-            requireRootPlugin()
-            apply(plugin = "kotlin")
+public class KotlinLibraryPlugin : InfrastructurePlugin() {
 
-            java {
-                targetCompatibility = JavaVersion.VERSION_1_8
-                sourceCompatibility = JavaVersion.VERSION_1_8
-            }
+    override fun Project.configure() {
+        apply(plugin = "kotlin")
 
-            kotlin.explicitApi()
-
-            configureKotlin()
-            configureKotlinTest()
-            configureRepositories()
-            configureKotlinDependencies()
-            configureKotlinTestDependencies(redmadrobotExtension.test)
+        java {
+            targetCompatibility = JavaVersion.VERSION_1_8
+            sourceCompatibility = JavaVersion.VERSION_1_8
         }
+
+        kotlin.explicitApi()
+
+        val extension = redmadrobotExtension
+        configureKotlin()
+        configureKotlinDependencies(extension.kotlinVersion)
+        configureKotlinTest(extension.test)
+        configureKotlinTestDependencies(extension.kotlinVersion, extension.test)
+        configureRepositories()
     }
 }
 
+private fun Project.configureKotlinTest(options: TestOptions) {
```

Comment:
Now it is an extension to be consistent with other extensions in this file.
I thought about converting all extensions to class members but decided to keep it "as is" for now. this approach looks quite strange but reduces the nesting level.

It will be better to create separated PR with convertation of all extensions like this to class members.